Sources/StreamChatUI/ChatMessageList/Attachments/ChatMessageVoiceRecordingAttachmentListView+ItemViewModel.swift

```diff
@@ -0,0 +1,187 @@
+//
+// Copyright © 2023 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+import StreamChat
+import UIKit
+
+extension ChatMessageVoiceRecordingAttachmentListView {
+    /// The ViewModel that drives interactions and events for
+    internal class ItemViewPresenter: AudioPlayingDelegate {
```

Comment:
Why is it called presenter if the description mentions view model?
Also, we haven't been using presenters (or at least not that I am aware of) in the SDK.

Reply:
The name of the file should become `ItemViewPresenter` and I'm going to change it. Why we are using a Presenter(or in general an interaction object here) is because the VoiceRecordingView in contrast with most of our views needs to manage events coming from the AudioPlayer (in addition to the usual user interaction events). During the implementation of the view, I realised that the file grew too big to manage easily and I decided to split it in a ViewModel and the View. After a discussion with Nuno we decided to change the ViewModel into a Presenter.

Reply:
Some additional context @polqf. This was a ViewModel initially, but since it holds a reference to a view, I suggested that we changed the name to a Presenter since it resembles more a Presenter than a ViewModel. Either way, I also pointed out that concern that we don't follow this pattern right now elsewhere in the SDK, although I wish we did. 
Anyways, handling video player events directly in the view is a pain in the ass, so I can understand that for this case, we can make an expectation and already introduce a UI Logic layer. But, it makes sense to make it internal for now since we are not sure yet how customers would customize this.

Reply:
Ok, got your point. It makes sense to me for it to stay internal, though.

Reply:
Yeap the plan is to keep it internal until it's (if ever) needed to be exposed.